src/pexprs-outputRecipe.js

```diff
@@ -36,91 +28,95 @@ function getIntervalInfo(expr, grammarInterval) {
+  return [
+    'range',
+    getMetaInfo(this, grammarInterval),
+    this.from,
+    this.to
+  ];
 };
 
-pexprs.Param.prototype.outputRecipe = function(sb, formals, grammarInterval) {
-  sb.append('this.param(' + this.index + ')' + getIntervalInfo(this, grammarInterval));
+pexprs.Param.prototype.outputRecipe = function(formals, grammarInterval) {
+  return [
+    'param',
+    getMetaInfo(this, grammarInterval),
+    this.index
+  ];
 };
 
-pexprs.Alt.prototype.outputRecipe = function(sb, formals, grammarInterval) {
-  sb.append('this.alt(');
-  for (var idx = 0; idx < this.terms.length; idx++) {
-    if (idx > 0) {
-      sb.append(', ');
-    }
-    this.terms[idx].outputRecipe(sb, formals, grammarInterval);
-  }
-  sb.append(')' + getIntervalInfo(this, grammarInterval));
+pexprs.Alt.prototype.outputRecipe = function(formals, grammarInterval) {
+  return [
+    'alt',
+    getMetaInfo(this, grammarInterval)
+  ].concat(this.terms.map(function(term) {
+    return term.outputRecipe(formals, grammarInterval);
+  }));
 };
 
-pexprs.Extend.prototype.outputRecipe = function(sb, formals, grammarInterval) {
+pexprs.Extend.prototype.outputRecipe = function(formals, grammarInterval) {
   var extension = this.terms[0]; // [extension, orginal]
-  extension.outputRecipe(sb, formals, grammarInterval);
+  return extension.outputRecipe(formals, grammarInterval);
 };
 
-pexprs.Seq.prototype.outputRecipe = function(sb, formals, grammarInterval) {
-  sb.append('this.seq(');
-  for (var idx = 0; idx < this.factors.length; idx++) {
-    if (idx > 0) {
-      sb.append(', ');
-    }
-    this.factors[idx].outputRecipe(sb, formals, grammarInterval);
-  }
-  sb.append(')' + getIntervalInfo(this, grammarInterval));
+pexprs.Seq.prototype.outputRecipe = function(formals, grammarInterval) {
+  return [
+    'seq',
+    getMetaInfo(this, grammarInterval)
+  ].concat(this.factors.map(function(factor) {
+    return factor.outputRecipe(formals, grammarInterval);
+  }));
 };
 
 pexprs.Star.prototype.outputRecipe =
 pexprs.Plus.prototype.outputRecipe =
 pexprs.Opt.prototype.outputRecipe =
 pexprs.Not.prototype.outputRecipe =
-pexprs.Lex.prototype.outputRecipe = function(sb, formals, grammarInterval) {
-  sb.append('this.' + this.constructor.name.toLowerCase() + '(');
-  this.expr.outputRecipe(sb, formals, grammarInterval);
-  sb.append(')' + getIntervalInfo(this, grammarInterval));
+pexprs.Lex.prototype.outputRecipe = function(formals, grammarInterval) {
+  return [
+    this.constructor.name.toLowerCase(),
+    getMetaInfo(this, grammarInterval),
+    this.expr.outputRecipe(formals, grammarInterval)
+  ];
 };
 
-pexprs.Lookahead.prototype.outputRecipe = function(sb, formals, grammarInterval) {
-  sb.append('this.la(');
-  this.expr.outputRecipe(sb, formals, grammarInterval);
-  sb.append(')' + getIntervalInfo(this, grammarInterval));
+pexprs.Lookahead.prototype.outputRecipe = function(formals, grammarInterval) {
+  return [
+    'la',
```

Comment:
While we're at it, why don't we change this to `lookahead`? I don't think we're getting a big savings here :-)